otredis.go

```diff
@@ -2,34 +2,68 @@ package otredis
 
 import (
 	"context"
+	"fmt"
+	"strings"
 
 	"github.com/go-redis/redis"
-	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go"
 	"github.com/opentracing/opentracing-go/ext"
 )
 
-// WrapRedisClient adds opentracing measurements for commands and returns cloned client
-func WrapRedisClient(ctx context.Context, c *redis.Client) *redis.Client {
+func WrapWithOpenTracing(ctx context.Context, client *redis.Client) *redis.Client {
 	if ctx == nil {
-		return c
+		return client
 	}
 	parentSpan := opentracing.SpanFromContext(ctx)
 	if parentSpan == nil {
-		return c
+		return client
 	}
+	clientWithContext := client.WithContext(ctx)
+	opts := clientWithContext.Options()
+	clientWithContext.WrapProcess(process(parentSpan, opts))
+	clientWithContext.WrapProcessPipeline(processPipeline(parentSpan, opts))
+	return clientWithContext
+}
 
-	// clone using context
-	copy := c.WithContext(c.Context())
-	copy.WrapProcess(func(oldProcess func(cmd redis.Cmder) error) func(cmd redis.Cmder) error {
+func process(parentSpan opentracing.Span, opts *redis.Options) func(oldProcess func(cmd redis.Cmder) error) func(cmd redis.Cmder) error {
+	return func(oldProcess func(cmd redis.Cmder) error) func(cmd redis.Cmder) error {
 		return func(cmd redis.Cmder) error {
-			tr := parentSpan.Tracer()
-			sp := tr.StartSpan("redis", opentracing.ChildOf(parentSpan.Context()))
-			ext.DBType.Set(sp, "redis")
-			sp.SetTag("db.method", cmd.Name())
```

Comment:
Seems like that line does not exist anymore, we should keep it in `doSpan`

Reply:
You mean the one with db.method?

Reply:
Yes

Reply:
Ok, fixed